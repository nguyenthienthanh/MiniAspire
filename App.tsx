import React, { useMemo } from 'react'
import { ActivityIndicator, useColorScheme } from 'react-native'

import { Provider as ReactReduxProvider } from 'react-redux'
import { PersistGate as ReduxPersistGate } from 'redux-persist/integration/react'
import './src/fakeApi'
import AppNavigator from './src/navigations/navigations'
import store, { persistor } from './src/stores'
import { ThemeProvider, Theme, colors } from 'react-native-elements'

const DefaultTheme: Theme = {
  ListItemTitle: {
    style: {
      fontSize: 13,
    },
  },
  ListItemSubtitle: {
    style: {
      fontSize: 12,
    },
  },
  colors,
}

const MyDarkTheme: Theme = {
  ...DefaultTheme,
  colors: {
    ...colors,
    primary: 'rgb(255, 45, 85)',
  },
}

const App = () => {
  const scheme = useColorScheme()

  const theme = useMemo(
    () => (scheme === 'dark' ? MyDarkTheme : DefaultTheme),
    [scheme],
  )

  return (
    <ReactReduxProvider store={store}>
      <ReduxPersistGate loading={<ActivityIndicator />} persistor={persistor}>
        <ThemeProvider theme={theme}>
          <AppNavigator />
        </ThemeProvider>
      </ReduxPersistGate>
    </ReactReduxProvider>
  )
}

export default App
