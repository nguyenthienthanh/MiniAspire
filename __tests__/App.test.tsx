/**
 * @format
 * @jest-environment jsdom
 */

import 'react-native'
import React from 'react'

import Enzyme, { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

import App from '../App'

describe('App', () => {
  it('should render correctly', () => {
    expect.assertions(1)

    const wrapper = shallow(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
