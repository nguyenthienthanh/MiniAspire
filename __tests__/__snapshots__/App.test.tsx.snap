// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    }
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "ListItemSubtitle": Object {
            "style": Object {
              "fontSize": 12,
            },
          },
          "ListItemTitle": Object {
            "style": Object {
              "fontSize": 13,
            },
          },
          "colors": Object {
            "black": "#242424",
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "grey": "rgba(0, 0, 0, 0.54)",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "grey": "#7d7d7d",
                "primary": "#007aff",
                "searchBg": "#dcdce1",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
            "white": "#ffffff",
          },
        }
      }
      useDark={false}
    >
      <AppNavigator />
    </ThemeProvider>
  </PersistGate>
</Provider>
`;
